extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -0,0 +1,157 @@
+import org.slf4j.LoggerFactory;
+
+public class DatasourceOperations {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DatasourceOperations.class);
+  DataSource datasource;
+
+  public DatasourceOperations(DataSource datasource) {
+    this.datasource = datasource;
+  }
+
+  public void executeScript() {
+    ClassLoader classLoader = DatasourceOperations.class.getClassLoader();
+    try (Connection connection = borrowConnection();
+        Statement statement = connection.createStatement()) {
+      BufferedReader reader =
+          new BufferedReader(
+              new InputStreamReader(
+                  Objects.requireNonNull(classLoader.getResourceAsStream("h2/schema-v1-h2.sql")),
+                  UTF_8));
+      StringBuilder sqlBuffer = new StringBuilder();
+      String line;
+      while ((line = reader.readLine()) != null) {
+        line = line.trim();
+        if (!line.isEmpty() && !line.startsWith("--")) { // Ignore empty lines and comments
+          sqlBuffer.append(line).append("\n");
+          if (line.endsWith(";")) { // Execute statement when semicolon is found
+            String sql = sqlBuffer.toString().trim();
+            try {
+              int rowsUpdated = statement.executeUpdate(sql);
+              LOGGER.debug("Query {} executed {} rows affected", sql, rowsUpdated);
+            } catch (SQLException e) {
+              LOGGER.error("Error executing query {}", sql, e);
+            }
+            sqlBuffer.setLength(0); // Clear the buffer for the next statement
+          }
+        }
+      }
+    } catch (IOException e) {
+      LOGGER.error("Error reading the script file", e);
+      throw new RuntimeException(e);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing the script file", e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  public <T> List<T> executeSelect(String query, Class<T> targetClass) {
+    try (Connection connection = borrowConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet s = statement.executeQuery(query);
+      List<T> results = ResultSetToObjectConverter.convert(s, targetClass);
+      return results.isEmpty() ? null : results;
+    } catch (Exception e) {
+      LOGGER.error("Error executing query {}", query, e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  public int executeUpdate(String query) {
+    try (Connection connection = borrowConnection();
+        Statement statement = connection.createStatement()) {
+      return statement.executeUpdate(query);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing query {}", query, e);
+      return -1;
+    }
+  }
+
+  public int executeUpdate(String query, Statement statement) throws SQLException {
+    System.out.println("Executing query in transaction : " + query);
+    try {
+      return statement.executeUpdate(query);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing query {}", query, e);
```

Comment:
We will need to either throw here, or catch `SQLException` and throw in its callers. Cannot just mute the exception.